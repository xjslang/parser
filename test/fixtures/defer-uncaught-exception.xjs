function testUncaughtException() {
  console.log('function start')
  defer {
    console.log('defer cleanup executed')
  }
  
  console.log('about to throw')
  throw new Error('uncaught error')
  console.log('this should not execute')
}

try {
  testUncaughtException()
} catch (e) {
  console.log('caught:', e.message)
}
