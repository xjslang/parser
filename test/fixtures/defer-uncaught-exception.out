function start
about to throw
defer cleanup executed
caught: uncaught error
